atc/api/jobserver/create_build.go

```diff
@@ -33,29 +32,10 @@ func (s *Server) CreateJobBuild(pipeline db.Pipeline) http.Handler {
 			return
 		}
 
-		scheduler := s.schedulerFactory.BuildScheduler(pipeline, s.externalURL, s.variablesFactory.NewVariables(pipeline.TeamName(), pipeline.Name()))
-
-		resourceTypes, err := pipeline.ResourceTypes()
-		if err != nil {
-			logger.Error("failed-to-get-resource-types", err)
-			w.WriteHeader(http.StatusInternalServerError)
-			return
-		}
-
-		versionedResourceTypes := resourceTypes.Deserialize()
-
-		resources, err := pipeline.Resources()
-		if err != nil {
-			logger.Error("failed-to-get-resources", err)
-			w.WriteHeader(http.StatusInternalServerError)
-			return
-		}
-
-		build, _, err := scheduler.TriggerImmediately(logger, job, resources, versionedResourceTypes)
+		build, err := job.CreateBuild()
```

Comment:
Shouldn't we make sure that this created build is marked 'triggered' so that it gets up-to-date resource versions? As discussed in https://github.com/concourse/concourse/issues/2438

It'd be nice if we kicked the scheduler loop here too. Fast is a feature.

Reply:
This already happens here: https://github.com/concourse/concourse/blob/ca9fd7c33fe63291ef291d40c4bd9ccc2ba692d4/atc/db/pipeline.go#L254-L255